src/lib.rs

```diff
@@ -93,6 +93,21 @@ pub fn is_socket(mode: u32) -> bool {
     type_bits(mode) == 0o014
 }
 
+/// Returns true if the set-user-ID bit is set
+pub fn has_setuid(mode: u32) -> bool {
```

Comment:
It looks like the tests fail because they're called `has_setuid` here but `is_setuid` in the tests. Let's go with `is_`, consistent with the other functions?

Reply:
(The idea was to keep them a bit separate from the is_{file,dir,…} lot. But if I couldn't keep up with myself, it probably wasn't a good idea.)